### PR TITLE
 Fix #7094: Back wall edge texture in water missing 

### LINKED ISSUE #7094: Back wall edge texture in water missing
**OS:** [Windows 8.1]
**Version:** [0.1.2]
**Commit/Build:** [9d04593] - I tried oldest build that is currently on openrct2.org download [0.1.2 build 9e72993] and it has same issue.

In water, the back wall edge is invisible. Screenshots will probably make it more clear:

Vanilla RCT2:
watertest

OpenRCT2:
watertestopenrct

- [ ] Reproducible in RCT2 (vanilla)?
- [x] Multiplayer?

**Steps to reproduce:**
1. Make high water
2. Place something behind it
3. See the something you placed

**Save game:**
[water test.zip](https://github.com/OpenRCT2/OpenRCT2/files/1671539/water.test.zip)

### PR DESCRIPTION
**Pull request by Gymnasiast**
First commit is a tiny refactor. Second commit contains the fix. Recommend viewing them one by one.

### CODE REVIEW

**Comment by Umdlye** [comment]
Sides are drawn wrong on one rotation, others are fine:

openrct2_2020-03-26_12-27-05

**Comment by Gymnasiast** [comment]
@Umdlye Hm, that's an original bug that resurfaced: https://github.com/OpenRCT2/OpenRCT2/issues/475

**Comment by Umdlye** [comment]
Some issues with back walls in oddly shaped water sections:

**Comment by Gymnasiast** [comment]
@Umdlye I think I have solved that one too :)

**Comment by Umdlye** [comment]
Looks good now! One small issue remains, was in the original game as well. Not sure if it's outside the scope of this PR.

openrct2_2020-03-26_13-31-25

**Comment by Gymnasiast** [comment]
Not sure if that's an issue/bug in the first place.

**Comment by pizza2004** [comment]
No issues here, it works how you would expect when you hide the vertical faces as well.

File: distribution/changelog.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 0.2.5+ (in development)
 ------------------------------------------------------------------------
+- Fix: [#475] GFX water sides drawn incorrectly.
```
Comment (Umdlye) [inline distribution/changelog.txt]: `- Fix: [#475] GFX water sides drawn incorrectly (original bug).` or `- Fix: [#475] Water sides drawn incorrectly (original bug).` 👼

**Comment by Gymnasiast** [comment]
>  was the fix mainly because of dl being incorrect?

That and the division by 8 instead of 16. Together they ensured no back wall was ever drawn.

**Pull request merged**